test/DeltaTriplesTest.cpp

```diff
@@ -0,0 +1,269 @@
+// Copyright 2023 - 2024, University of Freiburg
+//  Chair of Algorithms and Data Structures.
+//  Authors:
+//    2023 Hannah Bast <bast@cs.uni-freiburg.de>
+//    2024 Julian Mundhahs <mundhahj@tf.uni-freiburg.de>
+
+#include <gtest/gtest.h>
+
+#include "./util/GTestHelpers.h"
+#include "./util/IndexTestHelpers.h"
+#include "absl/strings/str_split.h"
+#include "engine/ExportQueryExecutionTrees.h"
+#include "index/DeltaTriples.h"
+#include "index/IndexImpl.h"
+#include "index/Permutation.h"
+#include "parser/RdfParser.h"
+
+namespace Matchers {
+inline auto InAllPermutations = [](auto InnerMatcher) {
+  return testing::AllOfArray(ad_utility::transform(
+      Permutation::ALL, [&InnerMatcher](const Permutation::Enum& perm) {
+        return testing::ResultOf(
+            absl::StrCat(".getLocatedTriplesPerBlock(",
+                         Permutation::toString(perm), ")"),
+            [perm](const DeltaTriples& deltaTriples) {
+              return deltaTriples.getLocatedTriplesPerBlock(perm);
+            },
+            InnerMatcher);
+      }));
+};
+inline auto NumTriplesInAllPermutations =
+    [](size_t expectedNumTriples) -> testing::Matcher<const DeltaTriples&> {
+  return InAllPermutations(AD_PROPERTY(LocatedTriplesPerBlock,
+                                       LocatedTriplesPerBlock::numTriples,
```

Comment:
```suggestion
  return InAllPermutations(AD_PROPERTY(LocatedTriplesPerBlock,
                                       numTriples,
```
(The point of this macro is to have the class name only once)